pkg/webhooks/controlplanemachineset/webhooks.go

```diff
@@ -62,15 +82,196 @@ var _ webhook.CustomValidator = &ControlPlaneMachineSetWebhook{}
 
 // ValidateCreate implements webhook.Validator so a webhook will be registered for the type.
 func (r *ControlPlaneMachineSetWebhook) ValidateCreate(ctx context.Context, obj runtime.Object) error {
+	var errs []error
+
+	cpms, ok := obj.(*machinev1.ControlPlaneMachineSet)
+	if !ok {
+		return errObjNotCPMS
+	}
+
+	controlPlaneMachines, err := r.fetchControlPlaneMachines(ctx)
+	if err != nil {
+		return fmt.Errorf("could not fetch existing control plane machines: %w", err)
+	}
+
+	// Ensure Control Plane Machine count matches the ControlPlaneMachineSet replicas
+	if cpms.Spec.Replicas == nil {
+		errs = append(errs, field.Required(field.NewPath("spec", "replicas"), "replicas field is required"))
+	} else if int(*cpms.Spec.Replicas) != len(controlPlaneMachines) {
+		errs = append(errs, field.Forbidden(field.NewPath("spec", "replicas"),
+			fmt.Sprintf("control plane machine set replicas (%d) does not match control plane machines (%d)", *cpms.Spec.Replicas, len(controlPlaneMachines))))
```

Comment:
Nit: Just an update on the wording please
```suggestion
			fmt.Sprintf("control plane machine set replicas (%d) does not match the current number of control plane machines (%d)", *cpms.Spec.Replicas, len(controlPlaneMachines))))
```